packages/graphiql-react/src/editor/hooks.ts

```diff
@@ -0,0 +1,102 @@
+import { Editor, EditorChange } from 'codemirror';
+import { RefObject, useEffect, useRef } from 'react';
+
+import onHasCompletion from './completion';
+
+export function useSyncValue(editor: Editor | null, value: string | undefined) {
```

Comment:
NIT: this could be me but I read this as a way to get a synchronous value rather than to synchronise a value with the editor instance